src/utils.py

```diff
@@ -83,6 +86,17 @@ def average_weights(w):
         w_avg[key] = torch.div(w_avg[key], len(w))
     return w_avg
 
+def fed_avg(local_weights, dataset_size_per_client):
+    avg_dict = {}
+    sum_dataset = sum(dataset_size_per_client)
+    for i, dictionary in enumerate(local_weights):
+        for key, tensor in dictionary.items():
+            if key not in avg_dict:
+                avg_dict[key] = tensor.clone() * dataset_size_per_client[i]/ sum_dataset
+            else:
+                avg_dict[key] += tensor.clone() * dataset_size_per_client[i]/sum_dataset
+    return avg_dict
+
```

Comment:
The FedAvg algorithm seems to be the one in the paper, so this is good. I am currently checking the accuracy on the Cifar 10 dataset. If you want to know what different options I am using, --model=cnn, --num_channels=3. Although, in the [https://arxiv.org/abs/1602.05629](paper) they mentioned around 280 rounds to achieve 80% accuracy on Cifar.